clap_complete/src/shells/zsh.rs

```diff
@@ -633,7 +633,7 @@ fn write_positionals_of(p: &Command) -> String {
         debug!("write_positionals_of:iter: arg={}", arg.get_id());
 
         let cardinality = if arg.is_multiple_values_set() || arg.is_multiple_occurrences_set() {
-            "*:"
+            "*::"
```

Comment:
Thanks for checking.

I was thinking a completion for trailing arguments usually doesn't need to know a previous positional argument (each argument having its own PossibleValues or ValueHint.). But given your concern I tried to come up with a counter example (not tested, but i think we can construct an app that my change breaks from this draft):

```
Command::new("explain_basic_cli_tools").arg(Arg::new("program").possible_value(["git", "cargo"]).required(true)).arg(Arg::new("args").multiple_values(true).value_hint(CommandWithArguments))
```
where we want users to call `explain_basic_cli_tools cargo example …` or `explain_basic_cli_tools git worktree add …` and artificially restrict possible commands.

For other value hints I may also introduce a regression if we have a file completer for a positional argument and the trailing arguments. then zsh's behaviour to let a file appear multiple times may get altered (didn't check, but i agree with you, from the generic point of view here in clap it's hard to say that either three colons or one/two colons is in general right).

fwiw I ended up in this rabbit hole looking at the completion of `rustup run nightly cargo bui<TAB>` where the colon is desired, otherwise we end up completing for a command `nightly` instead of `cargo`. too bad, i though this part was easy :weary: 

you don't happen to see a way out of this?

Reply:
I barely know anything about completions which makes reviewing them frustrating :)

We are working on https://github.com/clap-rs/clap/issues/3166 which should remove the complications of dealing with esoteric syntax in specific cases.  For external subcommands, we will probably have to provide a way for people to provide their own shell function much like we should add for our static completions today.